graph-pyramid-builder/src/main.py

```diff
@@ -0,0 +1,625 @@
+                                     cnames[ind[1]][0],
+                                     cnames[ind[0]][0],
+                                     CHUNK_SIZE,
+                                     CHUNK_SIZE,
+                                     int(np.mod(l_ind,f_info['rows'])),
+                                     int(l_ind/f_info['rows'])))
+            l_ind += 1
+        
+if __name__=="__main__":
+    """ Initialize argument parser """
+    logger.info("Parsing arguments...")
+    parser = argparse.ArgumentParser(prog='main', description='Build an image pyramid from data in a csv file.')
+
+    """ Define the arguments """
+    parser.add_argument('--inpDir',               # input image collection directory
+                        dest='inpDir',
+                        type=str,
+                        help='Path to input images.',
+                        required=True)
+    parser.add_argument('--outDir',               # Pyramid directory
+                        dest='outDir',
+                        type=str,
+                        help='The output directory for the flatfield images.',
+                        required=True)
+    
+    """ Get the input arguments """
+    args = parser.parse_args()
+
+    input_path = args.inpDir
+    output_path = Path(args.outDir)
+
+    logger.info('inpDir = {}'.format(input_path))
+    logger.info('outDir = {}'.format(output_path))
+
+    # Get the path to each csv file in the collection
+    input_files = [str(f.absolute()) for f in Path(input_path).iterdir() if ''.join(f.suffixes)=='.csv']
+
+    for f in input_files:
+        # Set the file path folder
+        folder = Path(f)
+        folder = folder.name.replace('.csv','')
+        logger.info('Processing: {}'.format(folder))
+        
+        # Load the data
+        logger.info('Loading csv: {}'.format(f))
+        data, cnames = load_csv(f)
+        column_names = data.columns
+        logger.info('Done loading csv!')
+
+        # Bin the data
+        logger.info('Binning data for {} features...'.format(column_names.size))
+        bins, bin_stats, linear_index = bin_data(data,column_names)
+        logger.info('Done!')
+        del data    # get rid of the original data to save memory
+
+        # Generate the default figure components
+        logger.info('Generating colormap and default figure...')
+        cmap = get_cmap()
+        fig, ax, data = get_default_fig(cmap)
+        logger.info('Done!')
+
+        # Generate the dzi file
+        logger.info('Generating pyramid metadata...')
+        info = metadata_to_graph_info(bins,output_path,folder)
+        logger.info('Done!')
+        
+        logger.info('Writing layout file...!')
+        write_csv(cnames,linear_index,info,output_path,folder)
+        logger.info('Done!')
+
+        # Create the pyramid
+        logger.info('Building pyramid...')
+        image = _get_higher_res(0,info,output_path,folder)
+    
```

Comment:
We also discussed adding in the ability to switch the "scale" of the axes from linear to log10, also, allowing it to be user defined outside of those two options might also be valuable. Let's at least implement the linear and log scale (which if you did above and I missed it I'm sorry) in such a way so that if later we want to go back and use a user defined function we can fairly easily.

Reply:
Log scaling would have to happen at the binning stage, which is easy enough to do.

Is there a preferred base I should use for log scaling?